options.go

```diff
@@ -62,6 +62,13 @@ func Auth(smtpUserName, smtpPasswd string) Option {
 	}
 }
 
+// UseLoginAuth sets LOGIN auth method
+func UseLoginAuth() Option {
```

Comment:
nit: it's better to drop Use prefix to be consistent with other options

Reply:
@umputun 
It is not that simple.
Look. Standard `net/smtp` library contains `PlainAuth()` function for PLAIN auth. By analogy, I made `LoginAuth()` function for LOGIN auth. So I can't use `LoginAuth()` name in options.
Please recommend how it is better to name the option method.

Reply:
Unless I'm missing something, I don't see  how the LoginAuth in your code related to the isolated Option method's name

Reply:
in this case, I'd rather call it `WithLoginAuth`

Reply:
There is no obvious need to export `LoginAuth()`. I propose to do the following.

1. Rename `LoginAuth()` (from `auth.go` file) to `newLoginAuth()` cuz this function is constructor for internal `loginAuth` struct that implements the smtp.Auth interface.

```go
func newLoginAuth(usr, pwd, host string) smtp.Auth {
	return &loginAuth{usr, pwd, host}
}

type loginAuth struct {
	user     string
	password string
	host     string
}
```

2. Use `LoginAuth()` name for option function.

@umputun  what do you think?

Reply:
fine with me. Btw, not sure why this constructor is even needed, maybe just for aesthetic reasons/symmetry

Reply:
It is needed more for aesthetic reasons.
It controls the set of required input parameters and cast the internal `loginAuth` type to `smtp.Auth` interface.

Reply:
I have refactored `auth.go`.

1. Now all that is related to authentication is the internal types and functions.
2. The option that sets the LOGIN auth method is now called `LoginAuth` to be consistent with other options.

Reply:
@umputun is it fine for you? May I resolve conversation?